### PR TITLE
Use new @clerk/nextjs package in with-clerk example, update Next.js authentication docs

### PR DESCRIPTION
**Pull request by colinclerk**
## Documentation / Examples

- [x] Make sure the linting passes

This PR makes two updates.

### 1. Update the `with-clerk` example to use a newly released `@clerk/nextjs` package

Previously, Clerk and Next.js required a combination of `@clerk/clerk-react` (for react) and `@clerk/clerk-sdk-node` (for api routes).  We combined the functionality into a single, Next.js-specific package to provide a better developer experience around installation and imports.

### 2. Update the Next.js Authentication docs to add an example of API route authentication

Currently, the authentication docs only provide examples for adding authentication to Pages (both with static generation and server-side rendering).  This PR adds an example for authenticating an API route.

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -103,6 +103,30 @@ export default Profile
...
+
+Authentication providers offer helpers to verify this secret and return authenticated user data to your API route.
+
+Let's look at an example that uses middleware to decorate the request object with user data.
+
+```js
+import { withSession } from '@clerk/nextjs/api'
+
+// Use withSession to decorate `req` with a session object
+export default withSession((req, res) => {
```
Comment (leerob) [inline docs/authentication.md:118]: Could we make this provider agnostic? We'd like to avoid picking favorites here 😄

Comment (colinclerk) [reply docs/authentication.md:118]: Yes, definitely.  In general, I tried to use the precedent from the `with-iron-session` examples above - talking generically about the pattern but leveraging a specific example.

I just added a commit that obscures the import:
https://github.com/vercel/next.js/pull/28906/commits/7a57cd445c8af25cb9ac7ce93d924bed788c5174

**Comment by colinclerk** [comment]
> It's still referencing Clerk below though in this section - and is this a valid code snippet from with-iron-session?

There's a catch-22 here.  Next.js doesn't provide a standardized interface for authentication providers to adhere to, so no two providers work alike.

So far, the Authentication docs have used generic copywriting about the available patterns, but specific examples from a single provider.  The single provider used for the first two examples is `with-iron-session`, for example:

> You can view this [example in action](https://next-with-iron-session.vercel.app/). Check out the [`with-iron-session`](https://github.com/vercel/next.js/tree/canary/examples/with-iron-session) example to see how it works.

https://nextjs.org/docs/authentication#authenticating-statically-generated-pages

In adding an example about Authenticating API routes, I used a Clerk example, but tried to use the precedent from `with-iron-session` to make sure I wasn't crossing the line re: picking favorites.  Specifically - I wrote generically about the pattern, but used a Clerk-specific example.

What do you think this is the best path forward here?

File: docs/authentication.md (hunk 1 above)
Comment (leerob) [reply docs/authentication.md:118]: It's still referencing Clerk below - I've tried to keep this provider agnostic, with iron session being an OSS library. Can we use that as the example and ensure the code snippet is reflected in the example?

Comment (leerob) [reply docs/authentication.md:118]: e.g. here https://github.com/vercel/next.js/blob/canary/examples/with-iron-session/pages/api/user.js

Comment (colinclerk) [reply docs/authentication.md]: Ok, just pushed an update with the linked example and re-ran linters.

Lmk if there are any other issues!

**Comment by braden-clerk** [comment]
Hey, taking this over! We switched the example to `with-iron-session`

Let me know if we need to make any more changes to this PR, thanks @leerob :)

**Comment by leerob** [comment]
If you'd like to get this merged faster, could we break up the example change and the docs change? The docs will still need some revisions.

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 132f95869c5e534436f6454018395039ef7943be

**test/integration/create-next-app/index.test.js** 
- create next app > should allow example with GitHub URL
- create next app > should allow example with GitHub URL and example-path
- create next app > should use --example-path over the file path in the GitHub URL
- create next app > should use npm as the package manager on supplying --use-npm with example

<details>
<summary>Expand output</summary>

  ● create next app › should allow example with GitHub URL

    Command failed with exit code 1 (EPERM): node /home/runner/work/next.js/next.js/packages/create-next-app/dist/index.js github-app --example https://github.com/vercel/next-learn-starter/tree/master/navigate-between-pages-starter

      136 |       ).toBeTruthy()
      137 |
    > 138 |       // Assert for dependencies specific to the typescript template
          |                         ^
      139 |       const pkgJSON = require(pkgJSONPath)
      140 |       expect(Object.keys(pkgJSON.dependencies)).toEqual([
      141 |         'next',

      at makeError (../node_modules/execa/lib/error.js:58:11)
      at handlePromise (../node_modules/execa/index.js:112:26)
      at integration/create-next-app/index.test.js:138:25
      at usingTempDir (integration/create-next-app/index.test.js:23:9)
      at Object.<anonymous> (integration/create-next-app/index.test.js:136:9)

  ● create next app › should allow example with GitHub URL and example-path

    Command failed with exit code 1 (EPERM): node /home/runner/work/next.js/next.js/packages/create-next-app/dist/index.js github-example-path --example https://github.com/vercel/next-learn-starter/tree/master --example-path navigate-between-pages-starter

      153 |   })
      154 |
    > 155 |   it('should allow example with GitHub URL', async () => {
          |                         ^
      156 |     await usingTempDir(async (cwd) => {
      157 |       const projectName = 'github-app'
      158 |       const res = await run(

      at makeError (../node_modules/execa/lib/error.js:58:11)
      at handlePromise (../node_modules/execa/index.js:112:26)
          at runMicrotasks (<anonymous>)
      at integration/create-next-app/index.test.js:155:25
      at usingTempDir (integration/create-next-app/index.test.js:23:9)
      at Object.<anonymous> (integration/create-next-app/index.test.js:153:9)

  ● create next app › should use --example-path over the file path in the GitHub URL

    Command failed with exit code 1 (EPERM): node /home/runner/work/next.js/next.js/packages/create-next-app/dist/index.js github-example-path-2 --example https://github.com/vercel/next-learn-starter/tree/master/navigate-between-pages-starter --example-path navigate-between-pages-starter

      172 |       expect(
      173 |         fs.existsSync(path.join(cwd, projectName, '.gitignore'))
    > 174 |       ).toBeTruthy()
          |                     ^
      175 |       expect(
      176 |         fs.existsSync(path.join(cwd, projectName, 'node_modules/next'))
      177 |       ).toBe(true)

      at makeError (../node_modules/execa/lib/error.js:58:11)
      at handlePromise (../node_modules/execa/index.js:112:26)
          at runMicrotasks (<anonymous>)
      at integration/create-next-app/index.test.js:174:25
      at usingTempDir (integration/create-next-app/index.test.js:23:9)
      at Object.<anonymous> (integration/create-next-app/index.test.js:172:9)

  ● create next app › should use npm as the package manager on supplying --use-npm with example

    Command failed with exit code 1 (EPERM): node /home/runner/work/next.js/next.js/packages/create-next-app/dist/index.js use-npm --use-npm --example https://github.com/vercel/next-learn-starter/tree/master/learn-starter

      326 |         '.eslintrc.json',
      327 |       ]
    > 328 |       files.forEach((file) =>
          |                         ^
      329 |         expect(fs.existsSync(path.join(cwd, file))).toBeTruthy()
      330 |       )
      331 |     })

      at makeError (../node_modules/execa/lib/error.js:58:11)
      at handlePromise (../node_modules/execa/index.js:112:26)
      at integration/create-next-app/index.test.js:328:25
      at usingTempDir (integration/create-next-app/index.test.js:23:9)
      at Object.<anonymous> (integration/create-next-app/index.test.js:326:9)

</details>

File: docs/authentication.md [hunk 2]
```diff
@@ -101,7 +101,7 @@ const Profile = ({ user }) => {
 export default Profile
 ```
 
-An advantage of this pattern is preventing a flash of unauthenticated content before redirecting. It's important to note fetching user data in `getServerSideProps` will block rendering until the request to your authentication provider resolves. To prevent creating a bottleneck and increasing your TTFB ([Time to First Byte](https://web.dev/time-to-first-byte/)), you should ensure your authentication lookup is fast. Otherwise, consider [static generation](#authenticating-statically-generated-pages).
+An advantage of this pattern is preventing a flash of unauthenticated content before redirecting. It's important to note fetching user data in `getServerSideProps` will block rendering until the request to your authentication provider resolves. To prevent creating a bottleneck and decreasing your TTFB ([Time to First Byte](https://web.dev/time-to-first-byte/)), you should ensure your authentication lookup is fast. Otherwise, consider [static generation](#authenticating-statically-generated-pages).
```
Comment (leerob) [inline docs/authentication.md:104]: "increasing" was correct here 👍

File: examples/with-clerk/.env.local.sample [hunk 3]
```diff
@@ -0,0 +1,2 @@
+NEXT_PUBLIC_CLERK_FRONTEND_API=your-frontend-api
```
Comment (leerob) [inline examples/with-clerk/.env.local.sample:1]: Could this be .example instead of sample? That's the typical format.

**Review by leerob — COMMENTED** [review]
Looking great!

File: docs/authentication.md (hunk 2 above)
Comment (colinclerk) [reply docs/authentication.md]: Yup, sorry - restored against an old canary and lost a commit. Fixed.

File: examples/with-clerk/package.json [hunk 4]
```diff
@@ -6,10 +6,9 @@
     "start": "next start"
   },
   "dependencies": {
-    "@clerk/clerk-react": "1.7.0",
-    "@clerk/clerk-sdk-node": "0.3.0",
-    "next": "10.0.9",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "@clerk/nextjs": "1.0.1",
+    "next": "11.1.2",
```
Comment (leerob) [inline examples/with-clerk/package.json:10]:
```suggestion
    "next": "latest",
```

**Review by leerob — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Pull request merged**